src/vllm_tgis_adapter/__main__.py

```diff
@@ -72,15 +89,42 @@ async def health() -> Response:
     return Response(status_code=200)
 
 
+if has_tokenization:
+    assert has_tokenization
+
+    @router.post("/tokenize")
+    async def tokenize(request: TokenizeRequest) -> JSONResponse:
+        generator = await openai_serving_tokenization.create_tokenize(request)  # noqa: F821
+        if isinstance(generator, ErrorResponse):
+            return JSONResponse(
+                content=generator.model_dump(),
+                status_code=generator.code,
+            )
+        assert isinstance(generator, TokenizeResponse)
+        return JSONResponse(content=generator.model_dump())
+
+    @router.post("/detokenize")
+    async def detokenize(request: DetokenizeRequest) -> JSONResponse:
+        generator = await openai_serving_tokenization.create_detokenize(request)  # noqa: F821
+        if isinstance(generator, ErrorResponse):
+            return JSONResponse(
+                content=generator.model_dump(),
+                status_code=generator.code,
+            )
+
+        assert isinstance(generator, DetokenizeResponse)
+        return JSONResponse(content=generator.model_dump())
+
+
 @router.get("/v1/models")
 async def show_available_models() -> JSONResponse:
-    models = await openai_serving_chat.show_available_models()
+    models = await openai_serving_completion.show_available_models()
     return JSONResponse(content=models.model_dump())
 
 
 @router.get("/version")
 async def show_version() -> JSONResponse:
-    ver = {"version": vllm.__version__}
+    ver = {"version": vllm.__version__, "commit": vllm.__commit__}
```

Comment:
This becomes slightly different than the upstream code, by also including the commit hash